packages/react-badge/src/components/PresenceBadge/PresenceBadge.types.ts

```diff
@@ -22,9 +22,11 @@ type PresenceBadgeCommons = {
    * @default false
    */
   outOfOffice: boolean;
+
+  strings: Record<PresenceBadgeStatus, string>;
```

Comment:
one thing to consider - we would require to provide all strings for each badge. Given there might be multiple badges on the page, each should get all texts and for each only one text will be used at a time, this requires consumers to either gather texts multiple times or create some advanced logic to reduce perf impact. Have you considered other alternatives that would require getting only texts that are being rendered?

Reply:
@jurokapsiar do you think the type should instead be `Partial<Record<PresenceBadgeStatus, string>>`? That would enable authors to only add a subset of strings, but at the detriment of providing warnings when strings are accidentally missed.

Reply:
Yeah, I added partial into the prop, I wasn't aware that the original approach would require all of the strings to be provided. With partial a user could override a single value

Reply:
we can just have one prop for the current string - the user knows what status they are setting, they can pair it with appropriate string.